### PR TITLE
some phrasing in the migration guides

### PR DESCRIPTION
**Pull request by 1ucian0**
Some changes in the migration guides about terminology around primitives (following the https://medium.com/qiskit/what-are-qiskit-primitives-9bf63c1eacc7 criteria)

### CODE REVIEW

File: docs/migration_guides/qi_migration.rst [hunk 1]
```diff
@@ -60,10 +60,10 @@ Contents
...
-    - *Aer Primitives* - The Sampler and Estimator in :mod:`qiskit_aer.primitives` --> ``from qiskit_aer.primitives import Sampler/Estimator``
-    - *Runtime Primitives* - The Sampler and Estimator in :mod:`qiskit_ibm_runtime` --> ``from qiskit_ibm_runtime import Sampler/Estimator``
-    - *Backend Primitives* - The BackendSampler and BackendEstimator in :mod:`qiskit.primitives` --> ``from qiskit import BackendSampler/BackendEstimator``
+    - *Reference Primitives* - The ``Sampler`` and ``Estimator`` in :mod:`qiskit.primitives` --> ``from qiskit.primitives import Sampler/Estimator``
+    - *Aer Primitives* - The ``Sampler`` and ``Estimator`` in :mod:`qiskit_aer.primitives` --> ``from qiskit_aer.primitives import Sampler/Estimator``
+    - *Qiskit Runtime Primitives* - The IBM's Qiskit Runtime primitive implementation. The ``Sampler`` and ``Estimator`` in :mod:`qiskit_ibm_runtime` --> ``from qiskit_ibm_runtime import Sampler/Estimator``
+    - *Backend Primitive* - Qiskit wrap for non-native backend primitives. The ``BackendSampler`` and ``BackendEstimator`` in :mod:`qiskit.primitives` --> ``from qiskit import BackendSampler/BackendEstimator``
```
Comment (javabster) [inline docs/migration_guides/qi_migration.rst:66]:
```suggestion
    - *Backend Primitives* - Qiskit wrappers for non-native backend primitives. The ``BackendSampler`` and ``BackendEstimator`` in :mod:`qiskit.primitives` --> ``from qiskit import BackendSampler/BackendEstimator``
```

File: docs/migration_guides/algorithms_migration.rst [hunk 2]
```diff
@@ -80,8 +80,8 @@ four different primitive implementations, depending on how you want to configure
 
     a. Using **local** statevector simulators for quick prototyping: **Reference Primitives** in :mod:`qiskit.primitives`
     b. Using **local** Aer simulators for finer algorithm tuning: **Aer Primitives** in :mod:`qiskit_aer.primitives`
-    c. Accessing backends using the **Qiskit Runtime Service**: **Runtime Primitives** in :mod:`qiskit_ibm_runtime`
-    d. Accessing backends using a **non-Runtime-enabled provider**: **Backend Primitives** in :mod:`qiskit.primitives`
+    c. Accessing backends using the **IBM's Qiskit Runtime service**: **Runtime primitives** in :mod:`qiskit_ibm_runtime`
```
Comment (ElePT) [inline docs/migration_guides/algorithms_migration.rst:83]: I would say either 
```suggestion
    c. Accessing backends using **IBM's Qiskit Runtime service**: **Runtime primitives** in :mod:`qiskit_ibm_runtime`
```

Comment (ElePT) [reply docs/migration_guides/algorithms_migration.rst:83]: or 
```suggestion
    c. Accessing backends using the **IBM Qiskit Runtime service**: **Runtime primitives** in :mod:`qiskit_ibm_runtime`
```

File: docs/migration_guides/algorithms_migration.rst [hunk 3]
```diff
@@ -80,8 +80,8 @@ four different primitive implementations, depending on how you want to configure
...
+    d. Accessing backends using a **non-native primitves in other provider**: **Backend Primitives** in :mod:`qiskit.primitives`
```
Comment (ElePT) [inline docs/migration_guides/algorithms_migration.rst:84]: I am not sure if this conveys well what the backend primitive does, maybe I don't understand it well. In this case, the backend primitives are the non-native primitive, right? but you are not using a "non-native primitive in other provider", unless you consider qiskit-terra a provider? In any case, the sentence looks a bit difficult to read, what about rephrasing what you wrote in the opflow migration section:

```suggestion
    d. Using a non-native primitive to access backends without a native primitive support: **Backend Primitives** in :mod:`qiskit.primitives`
```

File: docs/migration_guides/qi_migration.rst (hunk 1 above)
Comment (1ucian0) [reply docs/migration_guides/qi_migration.rst:66]: many wrappers?

Comment (ElePT) [reply docs/migration_guides/qi_migration.rst:66]: one per primitive, right? `BackendSampler` and `BackendEstimator`

File: docs/migration_guides/algorithms_migration.rst [hunk 4]
```diff
@@ -80,11 +80,11 @@ four different primitive implementations, depending on how you want to configure
 
     a. Using **local** statevector simulators for quick prototyping: **Reference Primitives** in :mod:`qiskit.primitives`
     b. Using **local** Aer simulators for finer algorithm tuning: **Aer Primitives** in :mod:`qiskit_aer.primitives`
-    c. Accessing backends using the **Qiskit Runtime Service**: **Runtime Primitives** in :mod:`qiskit_ibm_runtime`
-    d. Accessing backends using a **non-Runtime-enabled provider**: **Backend Primitives** in :mod:`qiskit.primitives`
+    c. Accessing backends using **IBM's Qiskit Runtime service**: **Runtime primitives** in :mod:`qiskit_ibm_runtime`
+    d. Accessing backends **without native primitves support**: **Backends without native primitive support** via :class:`~qiskit.primitives.BackendSampler` and :class:`~qiskit.primitives.BackendEstimator`.
```
Comment (ElePT) [inline docs/migration_guides/algorithms_migration.rst:84]: After our offline discussion, what about:

```suggestion
    d. Accessing backends **without native primitive support**: **Backend Primitives** (:class:`~qiskit.primitives.BackendSampler` and :class:`~qiskit.primitives.BackendEstimator`)
```

File: docs/migration_guides/algorithms_migration.rst (hunk 3 above)
Comment (1ucian0) [reply docs/migration_guides/algorithms_migration.rst]: As little as I like "backend primitives", I cannot find a better short description. So let's stay with it.

**Review by ElePT — COMMENTED** [review]
Thanks for the updates! I think that the phrasing is starting to look much clearer now.

File: docs/migration_guides/qi_migration.rst [hunk 5]
```diff
@@ -50,20 +50,23 @@ Contents
...
+
+    This guide uses the following naming convention:
 
-    - *Primitives* - Any Sampler/Estimator implementation
-    - *Reference Primitives* - The ``Sampler`` and ``Estimator`` in :mod:`qiskit.primitives` --> ``from qiskit.primitives import Sampler/Estimator``
-    - *Aer Primitives* - The ``Sampler`` and ``Estimator`` in :mod:`qiskit_aer.primitives` --> ``from qiskit_aer.primitives import Sampler/Estimator``
-    - *Qiskit Runtime Primitives* - The IBM's Qiskit Runtime primitive implementation. The ``Sampler`` and ``Estimator`` in :mod:`qiskit_ibm_runtime` --> ``from qiskit_ibm_runtime import Sampler/Estimator``
-    - *Backend Primitives* - Qiskit wrappers for non-native backend primitives. The ``BackendSampler`` and ``BackendEstimator`` in :mod:`qiskit.primitives` --> ``from qiskit import BackendSampler/BackendEstimator``
+    - *Primitives* - Any Sampler/Estimator implementation using base classes :class:`~qiskit.primitives.BackendSampler` and a :class:`~qiskit.primitives.BackendEstimator`.
+    - *Reference Primitives* -  :class:`qiskit.primitives.Sampler` and :class:`qiskit.primitives.Estimator` are reference implementation that come with Qiskit.
```
Comment (ElePT) [inline docs/migration_guides/qi_migration.rst:66]:
```suggestion
    - *Reference Primitives* -  :class:`qiskit.primitives.Sampler` and :class:`qiskit.primitives.Estimator` are reference implementations that come with Qiskit.
```

File: docs/migration_guides/qi_migration.rst [hunk 6]
```diff
@@ -50,20 +50,23 @@ Contents
...
 
-    - *Primitives* - Any Sampler/Estimator implementation
-    - *Reference Primitives* - The Sampler and Estimator in :mod:`qiskit.primitives` --> ``from qiskit.primitives import Sampler/Estimator``
-    - *Aer Primitives* - The Sampler and Estimator in :mod:`qiskit_aer.primitives` --> ``from qiskit_aer.primitives import Sampler/Estimator``
-    - *Runtime Primitives* - The Sampler and Estimator in :mod:`qiskit_ibm_runtime` --> ``from qiskit_ibm_runtime import Sampler/Estimator``
-    - *Backend Primitives* - The BackendSampler and BackendEstimator in :mod:`qiskit.primitives` --> ``from qiskit import BackendSampler/BackendEstimator``
+    - *Primitives* - Any Sampler/Estimator implementation using base classes :class:`~qiskit.primitives.BackendSampler` and a :class:`~qiskit.primitives.BackendEstimator`.
+    - *Reference Primitives* -  :class:`qiskit.primitives.Sampler` and :class:`qiskit.primitives.Estimator` are reference implementation that come with Qiskit.
+    - *Aer Primitives* - The `Aer <https://qiskit.org/ecosystem/aer>`_ primitive implemntations: class:`qiskit_aer.primitives.Sampler` and :class:`qiskit_aer.primitives.Estimator`.
```
Comment (ElePT) [inline docs/migration_guides/qi_migration.rst:67]:
```suggestion
    - *Aer Primitives* - The `Aer <https://qiskit.org/ecosystem/aer>`_ primitive implementations: class:`qiskit_aer.primitives.Sampler` and :class:`qiskit_aer.primitives.Estimator`.
```

File: docs/migration_guides/algorithms_migration.rst [hunk 7]
```diff
@@ -74,15 +74,17 @@ How to choose a primitive configuration for your algorithm
 
 *Back to* `TL;DR`_
 
-The classes in :mod:`qiskit.algorithms` state the base class primitive type (``Sampler``/``Estimator``)
-they require for their initialization. Once the primitive type is known, you can choose between
-four different primitive implementations, depending on how you want to configure your execution:
+The classes in
+:mod:`qiskit.algorithms` are initialised with any implementation of :class:`qiskit.primitive.BaseSampler` or class:`qiskit.primitive.BaseEstimator`.
```
Comment (ElePT) [inline docs/migration_guides/algorithms_migration.rst:78]: Tiny tiny detail, but the rest of the document follows American English spelling, so this should probably too:

```suggestion
:mod:`qiskit.algorithms` are initialized with any implementation of :class:`qiskit.primitive.BaseSampler` or class:`qiskit.primitive.BaseEstimator`.
```

File: docs/migration_guides/opflow_migration.rst [hunk 8]
```diff
@@ -21,11 +21,8 @@ the :mod:`~qiskit.opflow` module to the :mod:`~qiskit.primitives` and :mod:`~qis
 ..  attention::
 
     Most references to the :class:`qiskit.primitives.Sampler` or :class:`qiskit.primitives.Estimator` in this guide
-    can be replaced with instances of the:
-
-    - Aer primitives (:class:`qiskit_aer.primitives.Sampler`, :class:`qiskit_aer.primitives.Estimator`)
-    - Runtime primitives (:class:`qiskit_ibm_runtime.Sampler`, :class:`qiskit_ibm_runtime.Estimator`)
-    - Terra backend primitives (:class:`qiskit.primitives.BackendSampler`, :class:`qiskit.primitives.BackendEstimator`)
+    can be replaced with instances of any primitive implementation. For example Aer primitives (:class:`qiskit_aer.primitives.Sampler`/:class:`qiskit_aer.primitives.Estimator`) or IBM's Qiskit Runtime primitives (:class:`qiskit_ibm_runtime.Sampler`/:class:`qiskit_ibm_runtime.Estimator`).
+    Specific backends can be wrapped with (:class:`qiskit.primitives.BackendSampler`, :class:`qiskit.primitives.BackendEstimator`) to also present a primitive-compatible interfaces.
```
Comment (ElePT) [inline docs/migration_guides/opflow_migration.rst:25]: I like the term "primitive-compatible interfaces", I will note it down :)

**Review by ElePT — COMMENTED** [review]
I have gone through all of the changes one more time and found a few more small details. I think that after these are addressed the PR should be good to go.

File: docs/migration_guides/algorithms_migration.rst [hunk 9]
```diff
@@ -74,15 +74,17 @@ How to choose a primitive configuration for your algorithm
...
 
-    a. Using **local** statevector simulators for quick prototyping: **Reference Primitives** in :mod:`qiskit.primitives`
-    b. Using **local** Aer simulators for finer algorithm tuning: **Aer Primitives** in :mod:`qiskit_aer.primitives`
-    c. Accessing backends using the **Qiskit Runtime Service**: **Runtime Primitives** in :mod:`qiskit_ibm_runtime`
-    d. Accessing backends using a **non-Runtime-enabled provider**: **Backend Primitives** in :mod:`qiskit.primitives`
+Once the kind of primitive is known, you can choose between the primitive implementations that better adjust to your case. For example:
 
+    a. For quick prototyping, you can use the **reference implementations of primitives** included in Qiskit: :class:`qiskit.primitives.Sampler` and :class:`qiskit.primitives.Estimator`.
+    b. For finer algorithm tuning, a local simulator such as the **primitive implementation in Aer**: :class:`qiskit_aer.primitives.Sampler` and :class:`qiskit_aer.primitives.Estimator`.
```
[the remark this answers is not in the document]
Comment (1ucian0) [reply docs/migration_guides/algorithms_migration.rst:83]: Aer is an example of a local simulator that implement primitives.

File: docs/migration_guides/algorithms_migration.rst [hunk 10]
```diff
@@ -74,15 +74,17 @@ How to choose a primitive configuration for your algorithm
...
+    c. For executing in quantum hardware you can:
+
+       * access a native-primitive services, such as **IBM's Qiskit Runtime service** via :class:`qiskit_ibm_runtime.Sampler` and :class:`qiskit_ibm_runtime.Estimator`
```
[the remark this answers is not in the document]
Comment (1ucian0) [reply docs/migration_guides/algorithms_migration.rst]: I went for "services with native primitive implementations" in 40a3e699b6960c99b9dae42b9959041f93624a11

File: docs/migration_guides/algorithms_migration.rst (hunk 9 above)
[the remark this answers is not in the document]
Comment (ElePT) [reply docs/migration_guides/algorithms_migration.rst:83]: ok, never thought about it this way, but it makes sense

**Review by ElePT — APPROVED** [review]
LGTM! I have no more remarks on my side, only that I am not sure of why the docs build is failing. Maybe sth related to the sphinx theme update? My approval is probably not enough to merge so I will just approve now symbolically. Thanks again.

**Review by ElePT — APPROVED** [review]

**Review by mtreinish — APPROVED** [review]

**Pull request merged**